proposals/0184-dismiss-alert.md

```diff
@@ -7,35 +7,128 @@
 
 ## Proposed solution
 
-A new RPC is required to fix this use case. I propose to add an RPC called `DismissAlert`. This would dismiss any of an app's programmatically displayed alerts without needing to wait for the timeout to complete.
+A new RPC is required to fix this use case. I propose to add an RPC called `CancelInteraction`. This would dismiss any of an app's programmatically displayed alerts without needing to wait for the timeout to complete. This would affect `Alert`, `ScrollableMessage`, `Slider`, and `PerformInteraction`.
 
 ```xml
-<function name="DismissAlert" functionID="DismissAlertID" messagetype="request">
+<function name="CancelInteraction" functionID="DismissAlertID" messagetype="request" since="X.X.X">
+  <param name="cancelID" type="Integer" mandatory="false">
+    <description>
+      The ID of the specific interaction you want to dismiss. If not set, the most recent of the RPC type set in functionID will be dismissed.
+    </description>
+  </param>
+  <param name="functionID" type="Integer" mandatory="true">
+    <description>
+      The ID of the type of interaction the developer wants to dismiss. Only values 10, (PerformInteractionID), 12 (AlertID), 25 (ScrollableMessageID), and 26 (SliderID) are permitted.
+    </description>
+  </param>
+</function>
 
+<function name="CancelInteraction" functionID="DismissAlertID" messagetype="response">
+    <description>
+      If no applicable request can be dismissed, the result will be IGNORED.
+    </description>
+
+    <param name="success" type="Boolean" platform="documentation" mandatory="true">
+        <description> true if successful; false, if failed </description>
+    </param>
+        
+    <param name="resultCode" type="Result" platform="documentation" mandatory="true">
+        <description>See Result</description>
+        <element name="SUCCESS"/>
+        <element name="IGNORED"/>
+        <element name="INVALID_DATA"/>
+        <element name="OUT_OF_MEMORY"/>
+        <element name="TOO_MANY_PENDING_REQUESTS"/>
+        <element name="APPLICATION_NOT_REGISTERED"/>
+        <element name="GENERIC_ERROR"/>
+        <element name="REJECTED"/>
+        <element name="DISALLOWED"/>
+        <element name="UNSUPPORTED_RESOURCE"/>
+        <element name="WARNINGS"/>
+        <element name="ABORTED"/>
+    </param>
+        
+    <param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
+        <description>Provides additional human readable info regarding the result.</description>
+    </param>
 </function>
+```
 
-<function name="DismissAlert" functionID="DismissAlertID" messagetype="response">
+Additional Changes:
+
+```xml
+<function name="PerformInteraction" functionID="PerformInteractionID" messagetype="request">
+    <description>Triggers an interaction (e.g. "Permit GPS?" - Yes, no, Always Allow).</description>
```

Comment:
Should "no" be capitalized here?

Reply:
That's the exact description from `PerformInteraction` in the RPC spec. I just copied and pasted.